### PR TITLE
Bump bootkube v0.4.2: new pod-checkpointer + cloudprovider in bootstrap assets

### PR DESCRIPTION
**Pull request by aaronlevy**

### CODE REVIEW

File: modules/bootkube/resources/bootstrap-manifests/bootstrap-apiserver.yaml [hunk 1]
```diff
@@ -19,6 +19,7 @@ spec:
     - --authorization-mode=RBAC
     - --bind-address=0.0.0.0
     - --client-ca-file=/etc/kubernetes/secrets/ca.crt
+    - --cloud-provider=${cloud_provider}
```
Comment (Quentin-M) [inline modules/bootkube/resources/bootstrap-manifests/bootstrap-apiserver.yaml]: Won't compile. You need to add this variable to https://github.com/coreos/tectonic-installer/blob/a9b1e4c264d370679adc7a4d9432872b3032456c/modules/bootkube/assets.tf#L67.

File: modules/bootkube/resources/bootstrap-manifests/bootstrap-controller-manager.yaml [hunk 2]
```diff
@@ -12,6 +12,7 @@ spec:
     - controller-manager
     - --allocate-node-cidrs=true
     - --cluster-cidr=${cluster_cidr}
+    - --cloud-provider=${cloud_provider}
```
Comment (Quentin-M) [inline modules/bootkube/resources/bootstrap-manifests/bootstrap-controller-manager.yaml]: Won't compile. You need to add this variable to https://github.com/coreos/tectonic-installer/blob/a9b1e4c264d370679adc7a4d9432872b3032456c/modules/bootkube/assets.tf#L67.

**Comment by aaronlevy** [comment]
@Quentin-M updated the asset vars

**Comment by Quentin-M** [comment]
Test failure is unrelated.

**Pull request merged**